pyincore/baseanalysis.py

```diff
@@ -143,6 +144,24 @@ def set_output_dataset(self, id, dataset):
             # TODO handle error message
             return False
 
+    def validate_parameter_nested(self, parameter, parameter_spec):
+        is_valid = True
+        err_msg = ''
+
+        if type(parameter_spec['type']) is typing._GenericAlias:
```

Comment:
there is a warning about accessing a protected member _GenericAlias, is there any concern here of using this or is there a better way?